arithmetization/src/main/java/net/consensys/linea/zktracer/module/gas/Gas.java

```diff
@@ -19,61 +19,61 @@
 import java.nio.MappedByteBuffer;
 import java.util.List;
 
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Accessors;
 import net.consensys.linea.zktracer.ColumnHeader;
-import net.consensys.linea.zktracer.container.module.Module;
-import net.consensys.linea.zktracer.container.stacked.StackedList;
+import net.consensys.linea.zktracer.container.module.OperationSetModule;
+import net.consensys.linea.zktracer.container.stacked.StackedSet;
+import net.consensys.linea.zktracer.module.hub.Hub;
+import net.consensys.linea.zktracer.module.hub.defer.PostOpcodeDefer;
+import net.consensys.linea.zktracer.module.hub.fragment.common.CommonFragmentValues;
+import net.consensys.linea.zktracer.module.hub.signals.TracedException;
+import net.consensys.linea.zktracer.module.wcp.Wcp;
 import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.operation.Operation;
 
-public class Gas implements Module {
+@RequiredArgsConstructor
+@Accessors(fluent = true)
+public class Gas implements OperationSetModule<GasOperation>, PostOpcodeDefer {
   /** A list of the operations to trace */
-  private final StackedList<GasOperation> operations = new StackedList<>();
+  @Getter private final StackedSet<GasOperation> operations = new StackedSet<>();
 
-  // TODO: why a stackedList of GasOperation? It should be a StateLess module no ?
+  private CommonFragmentValues commonValues;
+  private GasCall gasCall;
+  private final Wcp wcp;
 
   @Override
   public String moduleKey() {
     return "GAS";
   }
 
-  @Override
-  public void enterTransaction() {
-    this.operations.enter();
-  }
-
-  @Override
-  public void popTransaction() {
-    this.operations.pop();
-  }
-
-  @Override
-  public int lineCount() {
-    return this.operations.lineCount();
-  }
-
   @Override
   public List<ColumnHeader> columnsHeaders() {
-    return null;
-  }
-
-  @Override
-  public void tracePreOpcode(MessageFrame frame) {
-    GasParameters gasParameters = extractGasParameters(frame);
-    this.operations.add(new GasOperation(gasParameters));
+    return Trace.headers(this.lineCount());
   }
 
-  private GasParameters extractGasParameters(MessageFrame frame) {
-    // TODO: fill it with the actual values
-    return new GasParameters(0, BigInteger.ZERO, BigInteger.ZERO, false, false);
+  public void call(GasCall gasCall, Hub hub, CommonFragmentValues commonValues) {
+    this.commonValues = commonValues;
+    this.gasCall = gasCall;
+    hub.defers().scheduleForPostExecution(this);
```

Comment:
I find VERy weird to schedule for post exec teh GAS module and not the GasOperation ... Of course it works but .... strange.

Reply:
How would you do it to make it look more natural?

Reply:
I don't see any beautiful and nice way to do it wo copying duplicate info and add cross-link ... Let's keep it like this :)